plugins/kubernetes/app/controllers/kubernetes_dashboard_controller.rb

```diff
@@ -0,0 +1,74 @@
+require 'kubeclient'
+
+class KubernetesDashboardController < ApplicationController
+  include CurrentProject
+
+  def index
+    render json: format_pod_info(get_pod_info).to_json
+  end
+
+  private
+
+  def get_pod_info
+    roles = hash_with(&default_role)
+    Environment.find(params[:environment]).cluster_deploy_groups.each do |cluster_deploy_group|
+      cluster_deploy_group.cluster.client.get_pods(
+          namespace: cluster_deploy_group.namespace,
+          label_selector: "project=#{current_project.name_for_label}").each do |pod|
+        labels = pod.metadata.labels
+        release = roles[labels.role_id][:deploy_groups]\
+                      [cluster_deploy_group.id][:releases][labels.release_id]
+        api_pod = Kubernetes::Api::Pod.new(pod)
+        release[:live_replicas] += 1 if api_pod.ready?
+      end
+    end
+    roles
+  end
+
+  def hash_with(*optional_params)
```

Comment:
I'm sure there is something clever going on here with blocks and lambdas, but it's very difficult to figure out what that is.